File: mobileApp/shared/src/commonMain/kotlin/com/leroybuliro/mobileapps/markets/domain/Product.kt
package com.leroybuliro.mobileapps.markets.domain

import org.jetbrains.compose.resources.DrawableResource

data class Product (
    val id: Int?,
    val image: ArrayList<DrawableResource>,
    val name: String,
    val description: String?,
    val category: String?,
    val subCategory: String?,
    val material: String?,
    val price: Double,
    val offerPrice: Double?,
    val stock: Int?,
)

File: services/inventory-service/internals/databases/models/product.go
package models

import (
	"gorm.io/gorm"
	"time"
)

type Product struct {
	ID          uint           `json:"id" gorm:"primaryKey"`
	Name        string         `json:"name" validate:"required" gorm:"not null"`
	Description string         `json:"description"`
	// Image       string         `json:"image" gorm:"text;not null;default:null"`
	Image       string         `json:"image" gorm:"text;default:null"`
	Price       float64        `json:"price" validate:"required,gt=0" gorm:"not null"`
	OfferPrice  float64        `json:"offer_price" validate:"gt=0" gorm:"not null;default:0"`
	Stock       int            `json:"stock" validate:"gte=0" gorm:"not null;default:0"`
	Category    string         `json:"category"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `json:"-" gorm:"index"`
}
